templates/returnExports.js

```diff
@@ -4,7 +4,8 @@
   } else if (typeof exports === 'object') {
     module.exports = factory(<%= cjs %>);
   } else {
-    root.<%= namespace %> = factory(<%= global %>);
+    var _root = root;
+    <% (Array.isArray(namespace) ? namespace : namespace.split(".")).forEach(function(name, index, list) { %><%if (index == (list.length - 1)) { %>_root["<%= name %>"] = factory(<%= global %>);<% } else { %>_root = (_root["<%= name %>"] = _root["<%= name %>"] || {});<% } %><% }); %>
```

Comment:
This kind of logic belongs in the module code, not the template. Please revisit if you are still interested in getting this merged.